### DISCUSSION TITLE
How to get relational data

### DISCUSSION THREAD

**Discussion by IzenDev** [supabase/supabase#12363] [Questions]
How can I get relational data when using `.stream()` in `Flutter` for `Real Time`

**Comment by TC3000** [comment]
This. Say I have three tables, `Post`, `Comment` and `User`. A post has a list of comments, and each comment belongs to a single user. How might I get a list of comments with their associated users as a stream? Say I want to build a reactive comment section where each comment is displayed together with the name and avatar of the user who wrote it. If a new comment is added, liked or removed, the comment section should react accordingly. 

I ask this because Supabase allow realtime queries for a single table only. This means I can't easily join my comment table with my user table.

**Reply by GaryAustin1** [reply]
You would need to make a feature request for flutter realtime streams.  But realtime itself is only on a single table so you have to use multiple subscriptions and do the joins with a select  in the payload handler.   You could do this in flutter with the subscription method currently.

**Reply by TC3000** [reply]
Can you please share an example? Three tables, `Post`, `Comment`, `User`, how to get all three as a stream.

**Reply by GaryAustin1** [reply]
I can't.  I don't use flutter.  I've not seen an example even in js.   I know it could be done by having multiple subscription handlers with a select using the standard embedded table joins.  The select would be called if any of the 3 table's subscriptions got a response to put everything together.  I do not know how https://supabase.com/docs/reference/dart/subscribe relates to generating a "stream".

**Reply by TC3000** [reply]
Alright, so from the link you provided, here's the code I found to listen to changes from a single table..

```dart
supabase.channel('public:countries').on(
  RealtimeListenTypes.postgresChanges,
  ChannelFilter(event: '*', schema: 'public', table: 'countries'),
  (payload, [ref]) {
    print('Change received: ${payload.toString()}');
  },
).subscribe();
```

I have a few questions:
1. Is it possible to listen to multiple, tables? (The only other option I found was to listen to all tables).
2. How do I make the joins. From where do I select columns from my `Post`, `Comment` and `User` tables?
3. Am I able to limit the number of comments to be returned, order them or paginate them?
4. How do I turn this into a stream that I can actually use in the UI?
